File: ios/Algo/Features/Auth/Models/LoginRequest.swift
// LoginRequest.swift
// Algo
//
// Request models for login and registration endpoints.

import Foundation

// MARK: - LoginRequest

/// Request body for POST /auth/login.
struct LoginRequest: Codable, Sendable {

    let email: String
    let password: String
    let deviceId: String
    let deviceType: String

    enum CodingKeys: String, CodingKey {
        case email
        case password
        case deviceId = "device_id"
        case deviceType = "device_type"
    }

    /// Creates a login request with the device type automatically set to "ios".
    init(email: String, password: String, deviceId: String) {
        self.email = email
        self.password = password
        self.deviceId = deviceId
        self.deviceType = "ios"
    }
}

// MARK: - RegisterRequest

/// Request body for POST /auth/register.
struct RegisterRequest: Codable, Sendable {

    let email: String
    let password: String
    let name: String
    let tenantId: String
    let deviceId: String
    let deviceType: String

    enum CodingKeys: String, CodingKey {
        case email
        case password
        case name
        case tenantId = "tenant_id"
        case deviceId = "device_id"
        case deviceType = "device_type"
    }

    /// Creates a registration request with the device type automatically set to "ios".
    init(email: String, password: String, name: String, tenantId: String, deviceId: String) {
        self.email = email
        self.password = password
        self.name = name
        self.tenantId = tenantId
        self.deviceId = deviceId
        self.deviceType = "ios"
    }
}

// MARK: - LogoutRequest

/// Request body for POST /auth/logout.
struct LogoutRequest: Codable, Sendable {

    let deviceId: String

    enum CodingKeys: String, CodingKey {
        case deviceId = "device_id"
    }
}

File: android/app/src/main/java/com/algonit/algo/features/auth/data/model/LoginRequest.kt
package com.algonit.algo.features.auth.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Request body for the POST /auth/login endpoint.
 */
@Serializable
data class LoginRequest(
    val email: String,
    val password: String,
    @SerialName("device_id") val deviceId: String,
    @SerialName("device_type") val deviceType: String = "android"
)

/**
 * Request body for the POST /auth/register endpoint.
 */
@Serializable
data class RegisterRequest(
    val email: String,
    val password: String,
    val name: String,
    @SerialName("tenant_id") val tenantId: String,
    @SerialName("device_id") val deviceId: String,
    @SerialName("device_type") val deviceType: String = "android"
)

/**
 * Request body for the POST /auth/api-key endpoint.
 */
@Serializable
data class ApiKeyLoginRequest(
    @SerialName("api_key") val apiKey: String,
    @SerialName("device_id") val deviceId: String,
)

/**
 * Request body for the POST /auth/logout endpoint.
 */
@Serializable
data class LogoutRequest(
    @SerialName("device_id") val deviceId: String
)

/**
 * Request body for the POST /auth/refresh endpoint.
 */
@Serializable
data class RefreshRequest(
    @SerialName("refresh_token") val refreshToken: String
)

/**
 * Generic status response for endpoints that return { "status": "ok" }.
 */
@Serializable
data class StatusResponse(
    val status: String
)
